core/lib/libtomcrypt/src/ciphers/sub.mk

```diff
@@ -1,5 +1,13 @@
 cflags-y += -Wno-unused-parameter
 
+ifeq ($(CFG_CRYPTO_AES_ARM64_CE),y)
+srcs-y += aes_arm64_ce.c
+cflags-aes_arm64_ce.c-y += -march=armv8-a+crypto
```

Comment:
On 15 June 2015 at 09:07, Jérôme Forissier <notifications@github.com> wrote:

> In core/lib/libtomcrypt/src/ciphers/sub.mk
> <https://github.com/OP-TEE/optee_os/pull/341#discussion_r32395540>:
>
> > @@ -1,5 +1,13 @@
> >  cflags-y += -Wno-unused-parameter
> >
> > +ifeq ($(CFG_CRYPTO_AES_ARM64_CE),y)
> > +srcs-y += aes_arm64_ce.c
> > +cflags-aes_arm64_ce.c-y += -march=armv8-a+crypto
>
> On Sun, Jun 14, 2015 at 10:40 AM, Ard Biesheuvel <notifications@github.com>
> wrote: > In core/lib/libtomcrypt/src/ciphers/sub.mk > <
> https://github.com/OP-TEE/optee_os/pull/341#discussion_r32377653>: > > >
> @@ -1,5 +1,13 @@ > > cflags-y += -Wno-unused-parameter > > > > +ifeq
> ($(CFG_CRYPTO_AES_ARM64_CE),y) > > +srcs-y += aes_arm64_ce.c > >
> +cflags-aes_arm64_ce.c-y += -march=armv8-a+crypto > > On 14 June 2015 at
> 07:36, Victor Chong <notifications@github.com> wrote: In
> core/lib/libtomcrypt/src/ciphers/sub.mk <
> https://github.com/OP-TEE/optee_os/pull/341#discussion_r32376595>: > @@
> -1,5 +1,13 @@ > cflags-y += -Wno-unused-parameter > > +ifeq
> ($(CFG_CRYPTO_AES_ARM64_CE),y) > +srcs-y += aes_arm64_ce.c >
> +cflags-aes_arm64_ce.c-y += -march=armv8-a+crypto -mgeneral-regs-only flag
> only covers platform-specific code? Will it include aes_arm64_ce.c? ​It
> should cover *all* code since otherwise, GCC may decide to spill data to
> FP/SIMD​ registers at any point in the program.
> Indeed, no C file is compiled without -mgeneral-regs-only. $ export
> CFG_ARM64_core=y $ export CROSS_COMPILE_core="ccache aarch64-linux-gnu-" $
> export PLATFORM=hikey $ make clean CLEAN . $ make V=1 | grep
> aarch64-linux-gnu-gcc | grep '\.c ' | grep -v general-regs-only $ make
> clean CLEAN . $ make V=1 | grep aarch64-linux-gnu-gcc | grep '\.c ' | grep
> general-regs-only | wc -l 244 Also, if it were the case we would get
> exceptions at runtime because due to CFG_WITH_VFP we disable VFP by default
> (that is, outside of
> thread_kernel_enable_vfp()/thread_kernel_disable_vfp()).
>
>
​OK, then we should be fine.

-- 
Ard.
​